models/user.go

```diff
@@ -877,6 +879,28 @@ func UpdateUser(u *User) error {
 	return updateUser(x, u)
 }
 
+// UpdateUserCols update user according special columns
+func UpdateUserCols(u *User, cols ...string) error {
+	// Organization does not need email
+	u.Email = strings.ToLower(u.Email)
+	if !u.IsOrganization() {
+		if len(u.AvatarEmail) == 0 {
+			u.AvatarEmail = u.Email
+		}
+		if len(u.AvatarEmail) > 0 {
+			u.Avatar = base.HashEmail(u.AvatarEmail)
+		}
+	}
+
+	u.LowerName = strings.ToLower(u.Name)
+	u.Location = base.TruncateString(u.Location, 255)
+	u.Website = base.TruncateString(u.Website, 255)
+	u.Description = base.TruncateString(u.Description, 255)
+
+	_, err := x.Id(u.ID).Cols(cols...).Update(u)
```

Comment:
The `updated_unix` column won't be updated. I think we need to do something similar to https://github.com/go-gitea/gitea/pull/2204